fdt.cc

```diff
@@ -1865,6 +1969,12 @@ device_tree::parse_dts(const string &fn, FILE *depfile)
 		}
 	}
 	collect_names();
+	// Return value indicates whether we've dirtied the tree or not and need to
+	// recollect names
+	if (garbage_collect_marked_nodes())
```

Comment:
This looks like it's a fairly expensive operation.  It would be nice to have a global flag on the fdt to see if any nodes have been parsed that contain the omit if unused flag, and then skip this if we're handling a dts that doesn't contain any (I'd expect this to be the common case).